src/main/java/org/jenkinsci/plugins/nomad/MigrationHelper.java

```diff
@@ -0,0 +1,328 @@
+                        getFieldValue(template, "cpu"),
+                        getFieldValue(template, "memory"),
+                        networks,
+                        devices
+                ),
+                new LogConfig(1, 10),
+                new Artifact[]{
+                        new Artifact(workerUrl, null, "/local/")
+                },
+                new Vault(getFieldValue(template, "vaultPolicies"))
+        );
+
+        TaskGroup taskGroup = new TaskGroup(
+                "jenkins-worker-taskgroup",
+                1,
+                new Task[]{task},
+                new RestartPolicy(0, 10000000000L, 1000000000L, "fail"),
+                new EphemeralDisk(getFieldValue(template, "disk"), false, false)
+        );
+
+        ConstraintGroup constraintGroup = new ConstraintGroup(getFieldValue(template, "constraints"));
+        List<Constraint> Constraints = constraintGroup.getConstraints();
+
+        Job job = new Job(
+                name,
+                name,
+                getFieldValue(template, "region"),
+                "batch",
+                getFieldValue(template, "priority"),
+                ((String) getFieldValue(template, "datacenters")).split(","),
+                Constraints,
+                new TaskGroup[]{taskGroup}
+        );
+
+        Gson gson = new GsonBuilder().setPrettyPrinting().disableHtmlEscaping().create();
+        JsonObject jobJson = new JsonObject();
+
+        jobJson.add("Job", gson.toJsonTree(job));
+
+        return gson.toJson(jobJson);
+    }
+
+    /**
+     * This is basically a copy of the old NomadApi#buildDriverConfig method (plugin version < 0.8.0).
+     */
+    private static Map<String, Object> buildDriverConfig(String name, String secret, String jenkinsUrl, String jenkinsTunnel,
+            NomadWorkerTemplate template) {
+        Map<String, Object> driverConfig = new HashMap<>();
+
+        String username = getFieldValue(template, "username");
+        Secret password = getFieldValue(template, "password");
+        if (username != null && !username.isEmpty()) {
+            Map<String, String> authConfig = new HashMap<>();
+            authConfig.put("username", username);
+            authConfig.put("password", password.getPlainText());
+
+            ArrayList<Map> credentials = new ArrayList<>();
+            credentials.add(authConfig);
+
+            driverConfig.put("auth", credentials);
+        }
+
+        ArrayList<String> args = new ArrayList<>();
+
+        if ("java".equals(getFieldValue(template, "driver"))) {
+            args.add("-jnlpUrl");
+
+            args.add(Util.ensureEndsWith(jenkinsUrl, "/") + "computer/" + name + "/worker-agent.jnlp");
+
+            // java -cp /local/slave.jar [options...] <secret key> <agent name>
+            if (!secret.isEmpty()) {
+                args.add("-secret");
+                args.add(secret);
+            }
+
+            driverConfig.put("jar_path", "/local/slave.jar");
```

Comment:
The jar_path must be a relative path. I guess this a bug in the current version.